### ISSUE TITLE
Provide rdfs:comment (and/or skos:definition or dct:description) text to all terms

### ISSUE THREAD

**Issue by lewismc** [ESIPFed/sweet#125]
Building on from #20 this issue simply aims to provide rdfs:comment (and/or skos:definition or dct:description) text to all terms.
Open tasks involve us collectively agreeing upon which vocabulary we wish to use e.g. rdfs:comment (and/or skos:definition or dct:description) and additionally whether we manually curate the comments or else automate this by fetching them from wikipedia/dbpedia/dictionary or elsewhere.

Any comments here?

**Comment by brandonnodnarb** [comment]
I have candidate term definitions for ~2K SWEET terms/classes pulled from Earth science glossaries we can sort through.  Although, I'm not sure the best way to do that at present.

**Comment by lewismc** [comment]
@brandonnodnarb where do they exists? Do you have them in electronic format somewhere?

At lunch, @dr-shorthair and I were discussing possibly just providing a _dct:description_ (although that would introduce a brand new namespace into SWEET) which is essentially a link to an alternate, _maintained_ description which exists elsewhere e.g. DBPedia. The keyword here is _maintained_. I think it would be a bad decision right now for us to go ahead and implement a whole bunch f descriptions which exists solely within SWEET. On the other hand, if they do link to other, better defined, maintained descriptions then it would make sense to link to them.

Any comments @brandonnodnarb ?

**Comment by rduerr** [comment]
What about definitions for terms that are defined in other ontologies (notably ENVO)?There are many ENVO terms that now use the GCW terminology definitions (though that document isn’t published yet).  It would be good to reference them directly, rather than reinvent the wheel.

Ruth

**Comment by lewismc** [comment]
I completely agree @rduerr

**Comment by brandonnodnarb** [comment]
@lewismc these are in a spreadsheet.  i'll see if I can clean it up and post it somewhere for review.

Also, I didn't think of this until you mentioned it, but another option could be to push these things to wikipedia/dbpedia, or make sure they are included and cited (and maintained) there.  Hmmm...let me think about this a bit.

**Comment by lewismc** [comment]
Yes ideally we could even get to this on Thursday as well. I thinl pusing to DBPedia would be an excellent idea. It would be excellent for us to re-use and/or make available as much of this to the wider audience. As this is a pretty large task, the best way may infact be the easiest way e.g. automating pulling comments from DBPedia. An example, very simple SPARQL query can be found as follows
```
prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> 
select distinct ?comment where {
  <http://dbpedia.org/resource/Gazetteer> rdfs:comment ?comment .
  FILTER (lang(?comment) = 'en')
}
```
Ofcourse we would merely substitute the subject IRI with whatever term we get from ESIP and then experiment with ```FILTER regex``` or other functions.
Thoughts?

**Comment by graybeal** [comment]
From a usability standpoint, an embedded description is much nicer (because it is there in front of you), and a little more confidence-inducing, because (a) it implies the author of the ontology (SWEET) vouch for it, (b) it is likely to be coherent with the purposes of the ontology, and (c) it is unlikely to drift without explicit reason. Add to that the opportunity for providing definitions that specifically disambiguate the term from its siblings, and fill the term space.

Some of these sources may achieve some of these goals. @lewismc, is your particular proposal to find the comments and 'bring them in', or simply to reference them in their original location? If the former, will the process be re-run every few years, or will we freeze this moment in our definitions?

Is it worth considering both a local copy of the definition and a reference to the source comment?

I'll be OK with whatever approach y'all think is reasonable and achievable. If there is decision-making involved, an ideal presentation would be a Google spreadsheet with the SWEET ontology name, term name, label, and external descriptions from whatever sources we are considering. That would make it easy to review the whole set at once as well as comment or vote on sources for particular terms, should it come to that.

**Comment by lewismc** [comment]
Hi @graybeal excellent questions, thanks for jumping in. You make some good points which I appreciate.

> is your particular proposal to find the comments and 'bring them in', 

If we did this, we would be essentially duplicating the content (and it would be appropriate to use of one rdfs:comment, skos:definition or dct:description). This is not to say that the things being represented are equals but merely that the way the thing is described is identical at that point in time. As you state, the actual literal values (from where they were acquired and where they exist within SWEET) will most likely diverge over time. Is this OK? It may be... but it may not be. More below...

> or simply to reference them in their original location? 

We _could_ also look into using ```rdfs:seeAlso``` as a mechanism for addressing the above issue (to clarify, this issue is the divergence of the content we bring in and encode as either rdfs:comment, skos:definition or dct:description AND the canonical source from which we obtained the information e.g. that same rdfs:comment over at DBPedia), where ```rdfs:seeAlso``` would reference the original source e.g. http://dbpedia.org/resource/Jet_Propulsion_Laboratory from which the rdfs:comment literal was extracted and rdfs:comment would be the actual literal content.
If this explanation is not clear then please let me know. An example would be as follows

Consider the following
https://github.com/ESIPFed/sweet/blob/eb8106314e4df9aae9042b792cfff749290bf8f3/src/reprDataProduct.ttl#L84-L87
Once the above work was done it would look as follows
```
###  http://sweetontology.net/reprDataProduct/Dataset
dprepr:Dataset rdf:type owl:Class ;
               rdfs:subClassOf dprepr:DataProduct ;
               rdfs:label "dataset"@en ;
               rdfs:comment "A  data set (or dataset, although this spelling is not present in many  contemporary dictionaries) is a collection of data. Most commonly a data  set corresponds to the contents of a single database table, or a single  statistical data matrix, where every column of the table represents a  particular variable, and each row corresponds to a given member of the  data set in question. The data set lists values for each of the  variables, such as height and weight of an object, for each member of  the data set. Each value is known as a datum. The data set may comprise  data for one or more members, corresponding to the number of rows."@en ;
               rdfs:seeAlso http://dbpedia.org/resource/Data_set .
```

If the former, will the process be re-run every few years, or will we freeze this moment in our definitions?

I'm not sure about this. We need to think it through.

**Comment by cmungall** [comment]
A bit confused by this discussion as I am used to ontologies where the definitions are authored by the developers of that ontology (sometimes adapted from an external source, with attribution). Randomly bringing in dictionary definitions could lead to incoherence, and how do we know the definitions reflects the intended meaning?

Yes, I have my handy blog post OntoTip entry for text definitions as well:
https://douroucouli.wordpress.com/2019/07/08/ontotip-write-simple-concise-clear-operational-textual-definitions/

Regardless of who writes them and what pipeline you use, it's super-important to track provenance of definitions, e.g. via axiom annotation

**Comment by graybeal** [comment]
OK, so who is the developer of SWEET these days? (Presumably the people who are currently maintaining it?) 

And how does that developer now create appropriate definitions, if not by referencing existing expertise?

**Comment by rduerr** [comment]
OK, so for the developer question, I do think ENVO's micro-citation is useful.  In other words, if I make a change to a term (any change) I annotate the change with my ORCID.  I also like using DBXREF's to cite the original source of the definitions.  I haven't looked at DBPedia; but perhaps that should be where I dump all the GCW terms and definitions?  

While I do like having embedded definitions, I really hate the idea of having to update the same definition in more than one place.  Would having them in DBPedia help with this problem?  

Also, I note that all the cryospheric terms and definitions and sources for those can be provided as csv file if that is helpful.

Thoughts?

**Comment by lewismc** [comment]
> Would having them in DBPedia help with this problem?
Yes it would, we would then look at the comment over in the DBPedia resource and determine whether we want a hard mapping. 

@cmungall thanks for chiming in. I agree with @graybeal here in that the answer to your statement

> ...where the definitions are authored by the developers of that ontology

That is essentially us. Raskin et al. never added simple labels or verbose descriptions so it is down to us to annotate and contextualize whatever we feel is necessary. 

IMHO DBPedia is the best resource I've come across where we can leverage existing knowledge. We can even do this one Class at a time with one pull request. Then every one of the proposed augmentation could be scrutinized.

Does this sound logical or is it way off?

**Comment by graybeal** [comment]
I'm wrestling with implications here, mostly because these external definitions are not versioned, are they? So please pardon my TLDR comments.

If we embed (copy) a definition we are then claiming it as our own, and ours won't track any changes to the original source (which may be the best thing); or if it does track changes, we'll have an ongoing monitoring task. In any case yes we'll need to evaluate each one.

If we link to definitions that live elsewhere, we still have the monitoring issue (what if that definition changes enough to make it wrong for SWEET?).  And if we make the link a hard one (along the lines of sameAs or exactMatch) we are effectively claiming it as our own, and therefore still have to track any changes made to the original to see if we agree. So we're effectively back at the first option.

I don't think we can support either of these approaches, even if we could create a great first version. And SWEET is not an authoritative real-world model that can be used for detailed reasoning about the world, and we can't pretend we will be able to come up with all-knowing definitions for these terms. It makes more sense to me to give people pointers to helpful information, and maintain SWEET as a relatively minimalist description of these earth science concepts.

So I think it would be best to have the definitions be notional, not authoritative. The relationship would then be 'notionallyDescribedBy', or better words to that effect, and there could be several of them, even with some contradictions between them. This best reflects the real world of SWEET in my opinion. 

With that approach they could be either embedded (with the definitions sourced in the provenance, and updated automatically from the original content); or referenced remotely (though that makes SWEET less handy to use). 

I'd prefer the embedded option, where multiple embedded definitions have been pulled from other sources (with date, source citation, and process citation). That follows best practices as far as I'm concerned.

**Comment by rduerr** [comment]
@pbuttigieg @cmungall Your take on this?

**Comment by dr-shorthair** [comment]
How about 
```
###  http://sweetontology.net/reprDataProduct/Dataset
dprepr:Dataset rdf:type owl:Class ;
               rdfs:subClassOf dprepr:DataProduct ;
               rdfs:label "dataset"@en ;
               skos:definition  [ 
                   rdfs:comment  "A  data set (or dataset, although this spelling is not present in many  contemporary dictionaries) is a collection of data. Most commonly a data  set corresponds to the contents of a single database table, or a single  statistical data matrix, where every column of the table represents a  particular variable, and each row corresponds to a given member of the  data set in question. The data set lists values for each of the  variables, such as height and weight of an object, for each member of  the data set. Each value is known as a datum. The data set may comprise  data for one or more members, corresponding to the number of rows."@en ;
                   dct:source <http://dbpedia.org/resource/Data_set> ;
                   dct:created "2019-08-16"^^xsd:date ;
             ] .
```
The range of `skos:definition` is `rdfs:resource`. This gets you the text locally along with the citation and the date it was copied. Of course the downside is that its now a property path `skos:definition/rdfs:comment` rather than just a simple property, but the complexity is not more than the problem being solved.

**Comment by lewismc** [comment]
This makes sense to me. 
I could easily code up something which opens a new pull request for every _hit_ that we get from DBPedia.
Let's see if we can get any more consensus...

**Comment by graybeal** [comment]
Ignoring the temptation to comment on the definition :-), I like this. Presumably there can be multiple definitions, which I think is helpful to prevent people from trying to "reason over the definitions" (or argue over the definitions, equally to the point). Good general-purpose definitions are very hard to build, so most aren't that good; the meaning is in the interplay of definitions.

In the interest of rigor, can the date be an ISO 8601 date+time+time zone? How does RDFS feel about (read: tolerate) that format?

@lewismc What about doing all the pull requests automatically in a branch, then push them all to a Google table (or similar) for each review/comment? (a) You don't want to give someone carpal tunnel approving pull requests, and (b) the likelihood should significantly favor acceptance, with a definition rejected only if there is an agreement it is clearly unacceptable or represents a different concept. (And in the former case, that it's just a poor definition, the disapproval could be represented by annotating the definition, rather than by not including it.)  Some system to keep track of the issues and rejections for future updates would be very helpful to minimize future maintenance costs.

I feel like

**Comment by cmungall** [comment]
Makes sense. I might go with wikidata rather than dbpedia. I also have code
to do wikidata matching.

See
https://github.com/EnvironmentOntology/envo/issues/833

We should request a SWEET ID property in wikidata see
https://www.wikidata.org/wiki/Property:P3859

**Comment by lewismc** [comment]
Hi folks, I'm going to start work on this issue. 

@cmungall I took a look at your provisional mappings from ENVO to Wikidata and they look excellent. If we could produce the same for SWEET it would be great. Can you point me at your implementation and I'll see what works needs done to adapt it for SWEET. Thank you

**Comment by lewismc** [comment]
@cmungall I created the following https://www.wikidata.org/wiki/Q76025584 but have a feeling it is incorrect.

**Comment by dr-shorthair** [comment]
If we want Wikidata to use SWEET internally, then it needs a CC-0 license …

**Comment by rduerr** [comment]
Well I think SWEET is now in the public domain - so would a public domain mark from the Creative Commons do?

**Comment by lewismc** [comment]
@dr-shorthair I'm not sure that's what we want... @cmungall can you please confirm?

@rduerr I don't think we need to be concerned with this.

**Comment by rduerr** [comment]
@lewismc Why not?  If SWEET is going to be heavily used it should have one or the other (CC0 or PDM).

**Comment by lewismc** [comment]
@rduerr I think SWEET may already heavily used...
Additionally, SWEET is software or a software artifact. CC is a public copyright license not a software license. I must admit, I've licensed content under CC-* but not software... I just wouldn't do it.
If you want to propose this to the wider audience then by all means go ahead. If you have buy in then please go ahead.

**Comment by cmungall** [comment]
In my experience, ontologies are licensed as content via some CC license or equivalent, not as software. Of course, I am strongly behind the idea that ontology engineering should be treated more like software engineering, and ontologies are computable artefacts. But nevertheless they are more like data than software. IMHO attempts to apply software licenses to ontologies have created confusion (worst is when people try and GPL them.. what does "linking" mean for an ontology?). 

I suggest a new issue for this however, as we have strayed from the original topic...

**Comment by cmungall** [comment]
> I took a look at your provisional mappings from ENVO to Wikidata and they look excellent. If we could produce the same for SWEET it would be great. I'm looking into https://github.com/cmungall/wikidata_ontomatcher right now.

Yes I will try and Dockerize with help from @wdduncan

> I created the following https://www.wikidata.org/wiki/Q76025584 but have a feeling it is incorrect.

Looks OK. Will you make a wikipedia page as well? The sync between these is still not totally clear to me

> If we want Wikidata to use SWEET internally, then it needs a CC-0 license

Correct. I support this but if there is reticence one option is to make a CC-0 subset for export. We do this for Mondo, where we don't export the text definitions.

**Comment by nicholascar** [comment]
Can I suggest that SWEET onts use `skos:definition` for anything it defines - all its classes, properties etc. - and the more generic `rdfs:comment` for other things like notes on imported ontology elements, the ontology-level metadata etc. This is the trend for current W3C ontology work like [DCAT 2](https://www.w3.org/TR/vocab-dcat-2/) & the [Profiles Vocabulary](https://w3c.github.io/dxwg/prof/).

**Comment by cmungall** [comment]
Seems reasonable. We use IAO for definitions in OBO but we probably should
have used skos.

IAO can be useful for more specific properties where rdfs:comment is too
broad - curator note, usage note, etc

**Comment by rduerr** [comment]
And this is why having real "best practices" for ontology development is a good idea.  If only some such agreement had been reached 8 months ago, I could have followed it!

**Comment by dr-shorthair** [comment]
As I [originally mentioned above](https://github.com/ESIPFed/sweet/issues/125#issuecomment-521815442) `skos:definition` is a sub-property of `skos:note`. 
It is an `rdf:Property` and `owl:AnnotationProperty` and has no global domain or range constraints. 
So the value can be a literal or a structured resource. 
So where the textual definition is sourced externally it is possible to retain the link with a citation. 

In the [example above](https://github.com/ESIPFed/sweet/issues/125#issuecomment-521815442) I suggested a blank-node, with an embedded text and source. Not fully happy about `rdfs:comment` for the text, but didn't want to stray outside the core RDF vocabularies. An alternative might be just to embed HTML markup, e.g. 
```
###  http://sweetontology.net/reprDataProduct/Dataset
dprepr:Dataset rdf:type owl:Class ;
               rdfs:subClassOf dprepr:DataProduct ;
               rdfs:label "dataset"@en ;
               skos:definition  """<p>A  data set (or dataset, although this spelling is not present in many  contemporary dictionaries) is a collection of data. Most commonly a data  set corresponds to the contents of a single database table, or a single  statistical data matrix, where every column of the table represents a  particular variable, and each row corresponds to a given member of the  data set in question. The data set lists values for each of the  variables, such as height and weight of an object, for each member of  the data set. Each value is known as a datum. The data set may comprise  data for one or more members, corresponding to the number of rows.</p>
<p><a href="http://dbpedia.org/resource/Data_set">accessed 2019-08-16T11:35:21.06Z from DBPedia</a></p>"""^^rdf:HTML ;
.
```

**Comment by cmungall** [comment]
as you likely predicted, I favor axiom annotations over embedding structure (especially HTML, which will likely be rendered with angle brackets in all tools)

it's very annoying owl has a separate reification vocabulary from rdf, but this is standard for anything built on an owl stack (including Protege)

Re: choice of skos. I forgot the annoying part that skos is undeclared w.r.t AP vs OP vs DP. This is potentially problematic (although I have yet to run into an issue with rdfs:comment, as there doesn't appear to be conflicting declarations in the same space, but this is not impossible). One advantage of IAO OM is that the definition property is unambiguously an AP

**Comment by dr-shorthair** [comment]
> skos is undeclared w.r.t AP vs OP vs DP

No - that part is OK I think. `skos:note` and descendants are explicitly AP. https://www.w3.org/TR/skos-reference/#L1738

**Comment by cmungall** [comment]
ah ok, great!

**Comment by lewismc** [comment]
New proposal
```
###  http://sweetontology.net/reprDataProduct/Dataset
dprepr:Dataset rdf:type owl:Class ;
               rdfs:subClassOf dprepr:DataProduct ;
               rdfs:label "dataset"@en ;
               skos:definition  [ 
                   rdfs:comment  "A  data set (or dataset, ..."@en ;
                   dct:source <http://dbpedia.org/resource/Data_set> ;
                   dct:created "2019-08-16T11:35:21.06Z"^^xsd:dateTimeStamp ;
             ] .
```

**Comment by rduerr** [comment]
Weighing in on the whole definitions thing.  I note that many individual SWEET term probably have a dozen or more concrete definitions in the scientific world, many of them quite disjoint.  In the SWEET/ENVO cryosphere mapping we've been doing we've been splitting the scientific definitions up into underlying concepts with definitions (which often doesn't agree with any of the definitions in the cryospheric glossaries) which came from work WMO/GCW funded.  This is all "cited" by associating the definition with the source, be it a glossary (or sometimes several) or a person.  This doesn't imply that dbpedia and wikidata definitions can't become definitive; but I question how the link to the scientific community(ies) and their definitions will be maintained (i.e., its a question of provenance).

**Comment by nicholascar** [comment]
> I question how the link to the scientific community(ies) and their definitions will be maintained 

As per discussion at the AGU workshop, the following guidance on definitional attribution was created: https://github.com/ESIPFed/sweet/wiki/SWEET-Class-Annotations-Proposal. There, the source of a definition is indicated with `dcterms:source` so a Wikipedia page, DOI or other article ref could be used.

**Comment by smrgeoinfo** [comment]
The syntax for the definitions using annotation with a source looks like a workable solution. It will surprise skos users who generally expect skos:definition to be text, but it is valid with the skos schema as @dr-shorthair pointed out above.

I think the more important point is one @rduerr brought up-  multiple disjoint or inconsistent but scientifically valid definitions associated with a term.  The key question is whether SWEET is a dictionary in which a 'word' can have multiple definitions, or a concept scheme, where each 'label' (word) is associated with a unique meaning (one definition for each 'word').   The AGI glossary already provides a widely used dictionary, with multiple definitions for words consistent with actual human usage. If multiple definitions are associated with a URI in SWEET, calling it an ontology would be misleading IMO.

If SWEET is intended as an artefact useful for information-interoperability, I think the concept scheme approach would be much more useful. This would require either allowing duplicate labels associated with distinct URIs for each unique concept, or a scheme for glossing labels to associate a unique label with each unique meaning.

**Comment by cmungall** [comment]
I just want to post the alternative proposal, to use owl reification:

the definition would look like this:

```
dprepr:Dataset rdf:type owl:Class ;
               rdfs:subClassOf dprepr:DataProduct ;
               rdfs:label "dataset"@en ;
               skos:definition  "A  data set (or dataset, although
               this spelling is not present in many  contemporary
               dictionaries) is a collection of data. Most commonly a
               data  set corresponds to the contents of a single
               database table, or a single  statistical data matrix,
               where every column of the table represents a
               particular variable, and each row corresponds to a
               given member of the  data set in question. The data set
               lists values for each of the  variables, such as height
               and weight of an object, for each member of  the data
               set. Each value is known as a datum. The data set may
               comprise  data for one or more members, corresponding
               to the number of rows."@en .


[ a owl:Axiom ;
  oboInOwl:hasDbXref <http://www.ncbi.nlm.nih.gov/pubmed/20973947> ;
  owl:annotatedProperty skos:definition ;
  owl:annotatedSource dprepr:Dataset;
  owl:annotatedTarget "A  data set (or dataset, although
               this spelling is not present in many  contemporary
               dictionaries) is a collection of data. Most commonly a
               data  set corresponds to the contents of a single
               database table, or a single  statistical data matrix,
               where every column of the table represents a
               particular variable, and each row corresponds to a
               given member of the  data set in question. The data set
               lists values for each of the  variables, such as height
               and weight of an object, for each member of  the data
               set. Each value is known as a datum. The data set may
               comprise  data for one or more members, corresponding
               to the number of rows."@en ;
  dct:source <http://dbpedia.org/resource/Data_set> ;
  dct:created "2019-08-16T11:35:21.06Z"^^xsd:dateTimeStamp 
] .
```

The obvious downside here is the repetition of the value. This would be terrible for editing ttl files in a text editor. However, it's not a problem if a standard ontology editing tool like Protege is used (aside: I have no idea what the normal edit environment is for sweet - in the OBO world almost no one hand edits owl/ttl files, most editing is done by non-technical people). In fact Protege has built-in support for axiom annotations, whereas editing blank nodes in Protege is sub-optimal.

Note that the syntactic repetition goes away if a native owl serialization is used, or if RDF* ever takes off.

An advantage of this scheme is that it can be applied to any triple, not just skos:definition. In this case skos doesn't explicitly prohibit the range from being an object rather than literal, but for other vocabularies there may be range constraints. Even when there is no explicit constraint, as @smrgeoinfo points out there are expectations/conventions.

It also simplifies things for the "99% use case" where you can query for simple definition strings over a single triple.

There is also a fair amount of tooling around axiom annotations. The owlapi understands them natively, tools like Protege support them. On the other hand, if you are working with raw rdf level APIs then reification can be a pain.

Both schemes involve blank nodes, but the blank nodes in the reification scheme "disappear" as the appropriate higher level of abstraction (rdf* or owl).

I am not pushing for one or the other, just charting the landscape of options.

The final decision will depend on many things I don't know the answer to: what does the edit workflow for sweet look like? Who will be doing the editing, what toolchains will they use? Is sweet's future more as an OWL ontology, or an RDFS ontology that utilizes a soupcon of OWL? What other kinds of axioms will have provenance attached in sweet? What kinds of queries are most important for users of sweet?

**Comment by lewismc** [comment]
Thanks @cmungall 
Most recently, as I've been doing large scale updates to SWEET, I've been using the text editor, grep, sed, awk, etc. most recently.
Usually, I load in Protege... reason being I always run a reasoner before I finish my editing.... in an attempt to prevent introduction of bugs. 

Regarding your last paragraph... honestly I think that most of this should be mentioned within the SWEET paper we are working on. If yoy are interested in working on reviewing that then please let me know. It is still in early development, so there is plenty of opportunity to influence,

**Comment by cmungall** [comment]
Yes, just email me a link!

Re: running reasoner - as discussed on another ticket you can use robot
reason, but as sweet is still quite 'weakly typed' finding actual explicit
incoherencies will be rare. You may also be interested in robot report.

>

**Comment by rduerr** [comment]
I agree with @smrgeoinfo that it makes more sense to me for SWEET to be a "concept scheme, where each 'label' (word) is associated with a unique meaning (one definition for each 'word'). "  But that implies, like ENVO already does, that concept labels be explicit.  For example, thermokarst to many geologists is a type of topography, while to a cryologist it usually is a landform (or do I have that backwards?).  So instead, ENVO defines thermokarst to be the landform and the process to be thermokarst formation process.  The question then becomes where does SWEET stop in its definitions - I don't think it should duplicate ENVO but they do need to be harmonized at some level.

This is fine from an ontological standpoint I think; but it does mean that someone using the ontology for data searches should either offer the choice of term that have thermokarst in them (like Ontobee does except in the data search case you are looking for a data set matching your query) or return everything matching the process form of the term as well as those matching the landform.

**Comment by lewismc** [comment]
Hi @cmungall @rduerr @nicholascar @smrgeoinfo @graybeal @dr-shorthair @brandonnodnarb 

[I'm nearly finished with some Scala code](https://github.com/ESIPFed/sweet-tools/pull/5) which will automate the process of augmenting a _defintion_ for each _owl:Class_. 

The question I have right now is as follows. wikidata uses [schema:description](https://schema.org/description)... therefore should we do the same?

**Comment by wdduncan** [comment]
@lewismc 
Does the wiki content look more like definitions or descriptions?  Can you post a query that gets a 100 or so records?

**Comment by lewismc** [comment]
@wdduncan thank you for chiming in. Yes absolutely!
```
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX schema: <http://schema.org/>
SELECT ?o WHERE {
	?s rdfs:label "experiment"@en .
    ?s schema:description ?o .
    FILTER(LANGMATCHES(LANG(?o), "en"))
    FILTER(STRLEN(?o) > 15)
}
```

https://query.wikidata.org/sparql?query=PREFIX%20rdfs%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0APREFIX%20schema%3A%20%3Chttp%3A%2F%2Fschema.org%2F%3E%0ASELECT%20%3Fo%20WHERE%20%7B%0A%09%3Fs%20rdfs%3Alabel%20%22experiment%22%40en%20.%0A%20%20%20%20%3Fs%20schema%3Adescription%20%3Fo%20.%0A%20%20%20%20FILTER(LANGMATCHES(LANG(%3Fo)%2C%20%22en%22))%0A%20%20%20%20FILTER(STRLEN(%3Fo)%20%3E%2015)%0A%7D

This returns the `schema:description` for the `rdfs:label` as **"scientific procedure carried out to support, refute, or validate a hypothesis"**

**Comment by lewismc** [comment]
... the other option is that we include all `schema:description`'s simply as `schema:description`'s. If this is good enough for wikidata is it good enough for SWEET?

**Comment by lewismc** [comment]
If we can reach some agreement on the above then I will be ready to push the source code and create a pull request with this information.

**Comment by wdduncan** [comment]
In this case, it looks more like a definition to me. If we go with this, I suppose we can change others to descriptions as we find them?

**Comment by wdduncan** [comment]
Would it be worthwhile to put the terms and `schema:descriptions` in a spreadsheet for us to review? The we could mark if the annotation is a description or definition?

**Comment by lewismc** [comment]
> Would it be worthwhile to put the terms and schema:descriptions in a spreadsheet for us to review?

I could do this... however mapping from those determinations back to the correct logic for the pull request cannot be automated. We are talking about thousands of labels for which we are looking to obtain definitions. I completely understand why manual curation is your desire... but once that spreadsheet is populated, then I am left to manually map all of those decisions back into source code... that would take me literally years...
I kinda torn here...

**Comment by wdduncan** [comment]
Maybe a statistical sampling would work?
Perhaps you could put the content in a triple store, and then use a delete/update query to change the annotations?

**Comment by lewismc** [comment]
Thanks for the suggestions @wdduncan. It's looking more and more like this is not going to be a reproduceable process... which is what I would have preferred. 
I'll work on pulling together the pull request first (we can always close it without merging into  master) such that we can evaluate the size of the peer review process. 
After that I'll [revisit your comment](https://github.com/ESIPFed/sweet/issues/125#issuecomment-598849480).

**Comment by smrgeoinfo** [comment]
Seems like we could put the schema:description content from wikidata in rdfs:comment, and then wouldn't have to import another big namespace.  Especially considering that the content of wikidata descriptions is likely to be pretty heterogeneous.  If we're going to add definitions, I'd prefer using skos:definition (already used in realmHydroBody.ttl, only two occurrences though).   A

**Comment by dr-shorthair** [comment]
Reviving this issue following the SemTech meeting today: 

here is a preliminary pattern for coordinating labels and definitions harvested from multiple external sources - 
```
###  http://sweetontology.net/reprDataProduct/Dataset
dprepr:Dataset rdf:type owl:Class ;
               rdfs:subClassOf dprepr:DataProduct ;
               rdfs:label "dataset"@en ; # Up to here, this is from SWEET, the following is pulled from alternative external sources
               skos:definition  [ 
                   skos:definition  "A  data set (or dataset, although this spelling is not present in many  contemporary dictionaries) is a collection of data. Most commonly a data  set corresponds to the contents of a single database table, or a single  statistical data matrix, where every column of the table represents a  particular variable, and each row corresponds to a given member of the  data set in question. The data set lists values for each of the  variables, such as height and weight of an object, for each member of  the data set. Each value is known as a datum. The data set may comprise  data for one or more members, corresponding to the number of rows."@en ;
                   skos:prefLabel "Data set"@en ;
                   dcterms:source <http://dbpedia.org/resource/Data_set> ;
                   dct:created "2019-08-16"^^xsd:date ;
             ]  ;
               skos:definition  [ 
                   skos:definition "collection of data"@en ;
                   skos:prefLabel "data set"@en ;
                   dcterms:source <https://www.wikidata.org/wiki/Q1172284> ;
                   dct:created "2021-08-25"^^xsd:date ;
             ]  ;
               skos:definition  [ 
                   skos:definition  "A collection of data, published or curated by a single agent, and available for access or download in one or more representations"@en ;
                   skos:prefLabel "Dataset"@en ;
                   dcterms:source <https://www.w3.org/TR/vocab-dcat/#Class:Dataset> ;
                   dct:created "2021-08-25"^^xsd:date ;
             ]  ;
.
```

The pattern above is not necessarily the ultimate solution. But it records separate labels and definitions - here embedded in blank-nodes - clearly identified by source, so data formatted this way could be transformed to another pattern with SPARQL queries if another pattern is preferred. 

@rduerr does this help?

**Comment by rduerr** [comment]
Yup - absolutely!!!

**Comment by rduerr** [comment]
See discussion [https://github.com/ESIPFed/sweet/discussions/259](https://github.com/ESIPFed/sweet/discussions/259l) for a proposal on this topic.

As for blank nodes with these definition blocks, anyone want to tackle a proposal for that?  I'd like to move forward on this stuff!!!

**Comment by lewismc** [comment]
If someone can provide me with the proposed data model I can write the
solution. I stepped away from this because we were still unsure about the
data model.
Did we come to a consensual agreement?
Thanks
-- 

*Lewis*
Dr. Lewis J. McGibbney Ph.D, B.Sc
*Skype*: lewis.john.mcgibbney
